src/cata_tiles.cpp

```diff
@@ -1228,13 +1327,25 @@ bool cata_tiles::draw_sprite_at(const std::vector<int> &spritelist, int x, int y
     destination.w = tile_width;
     destination.h = tile_height;
 
+    if ( svlist.empty() ) {
+        // render nothing
+        return true;
+    }
+
+    const std::vector<int> *spritelist = svlist.pick(loc_rand);
```

Comment:
You don't check for NULL here, which is fine as the loading makes the sure each tile has at least one sprite list, so `pick` should never return NULL. But should assign the result to a reference. This allows to get rid of the changed member access: `.foo` to `->foo`.

Additionally: what's with all the `at` calls? I see no catch of `std:.out_of_range`, and the index is checked anyway, so it should never throw anyway. Why not use the `operator[]`?